### PR TITLE
Inspector web gtm beta

### PR DESCRIPTION
**Pull request by thoragudf**

### CODE REVIEW

**Comment by hugihlynsson** [comment]
#### 📝 Changed routes:


| [**…nnect-inspector-to-gtm**](https://docs-3tz9ev8ug.avo-dev.app/inspector/connect-inspector-to-gtm) |
|:-:|
|<a href="https://docs-3tz9ev8ug.avo-dev.app/inspector/connect-inspector-to-gtm">Screenshot of /inspector/connect-inspector-to-gtm</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/4g3e-2hQ-Yx_SKrsJDAHzGoGX22FUxsNL9j_7V3bO5kLguB1SpzvKD-5Hx9VQq5qULf-y7l0UzMJz3tIjNe4eA.png">(view full size)</a> |







Commit bb76d5434ad949789406669f90326beda40518e7 (https://docs-3tz9ev8ug.avo-dev.app).

File: pages/inspector/connect-inspector-to-gtm.mdx [hunk 1]
```diff
@@ -1,14 +1,70 @@
...
+
+<Callout type="info" emoji="💡">
+We recommend starting by installing the Inspector on your development or testing GTM container and connect it to the Inspector dev environment. That is both so your Inspector production environment is not contaminated by testing data and because the Inspector development environment is designed to handle low data volumes real time while the production environment is optimized for large amount of production data and has 1 - 2 hour delay.
+</Callout>
+
+### Importable GTM recipe
+
+To add observability to all data layer events with our importable GTM recipe [in under 90 seconds](https://www.loom.com/share/c659b857d4d947729683bf11691f441e), follow the steps below:
+1. [Download our GTM recipe here]()
+2. Import the recipe into your GTM container.
```
Comment (thoragudf) [inline pages/inspector/connect-inspector-to-gtm.mdx:45]: Would be helpful to include a screenshot here and perhaps more details in order to make this foolproof.

File: pages/inspector/connect-inspector-to-gtm.mdx [hunk 2]
```diff
@@ -1,14 +1,70 @@
...
+<Callout type="info" emoji="💡">
+We recommend starting by installing the Inspector on your development or testing GTM container and connect it to the Inspector dev environment. That is both so your Inspector production environment is not contaminated by testing data and because the Inspector development environment is designed to handle low data volumes real time while the production environment is optimized for large amount of production data and has 1 - 2 hour delay.
+</Callout>
+
+### Importable GTM recipe
+
+To add observability to all data layer events with our importable GTM recipe [in under 90 seconds](https://www.loom.com/share/c659b857d4d947729683bf11691f441e), follow the steps below:
+1. [Download our GTM recipe here]()
+2. Import the recipe into your GTM container. 
+3. Add the source specific Inspector API key.
```
Comment (thoragudf) [inline pages/inspector/connect-inspector-to-gtm.mdx:46]: A screenshot and more details would be helpful here. 

Something "Navigate to the Inspector Setup tab of your Inspector source, and copy the API key. Then paste it [where I shoould paste it]"

File: pages/inspector/connect-inspector-to-gtm.mdx [hunk 3]
```diff
@@ -1,14 +1,70 @@
...
+We recommend starting by installing the Inspector on your development or testing GTM container and connect it to the Inspector dev environment. That is both so your Inspector production environment is not contaminated by testing data and because the Inspector development environment is designed to handle low data volumes real time while the production environment is optimized for large amount of production data and has 1 - 2 hour delay.
+</Callout>
+
+### Importable GTM recipe
+
+To add observability to all data layer events with our importable GTM recipe [in under 90 seconds](https://www.loom.com/share/c659b857d4d947729683bf11691f441e), follow the steps below:
+1. [Download our GTM recipe here]()
+2. Import the recipe into your GTM container. 
+3. Add the source specific Inspector API key. 
+4. When testing, adjust the the Avo Inspector Init tag by changing the 'prod' value to 'dev' on the inspector.__ENV__ key.
```
Comment (thoragudf) [inline pages/inspector/connect-inspector-to-gtm.mdx:47]: Screenshot and more info on where to find that key in the tag would be helpful here

File: pages/inspector/connect-inspector-to-gtm.mdx [hunk 4]
```diff
@@ -1,14 +1,70 @@
...
+5. Publish the container.
+6. Visit the Avo Inspector dashboard in one hour to see
```
Comment (thoragudf) [inline pages/inspector/connect-inspector-to-gtm.mdx:49]: Should probably specify dev if testing and prod if in production

File: pages/inspector/connect-inspector-to-gtm.mdx [hunk 5]
```diff
@@ -1,14 +1,74 @@
...
+
+<Callout type="info" emoji="💡">
+The Initialization code is already included in a tag, so you just need to provide the Inspector API GTM variable the source specific API key, which you can find in your Avo source Inspector setup. 
+</Callout>
+
+### Inspector Get Schema function
+
+Another option is to initialize Inspector via creating your own HTML tag and using the HTML script you can find in the source Inspector setup.
+
+To add observability to single events with our Inspector Get Schema function, create a HTML script tag for each event you want to add observability to. The steps are shown in the screen shot below and make sure to use the same trigger and variables as you are using for the event you're dispatching to your analytics vendor.
```
Comment (thoragudf) [inline pages/inspector/connect-inspector-to-gtm.mdx:60]: Are you referring to this screenshot https://uploads-ssl.webflow.com/5ec440af46599341a9a10225/64ad6294b6487fc8fa322a81_zgjampslJbIiIlKFqqkYJbs65C5_tBuvKQWF7A72E-SOgz4uk0lwZi6ZAN4OOFna2DjmH94Vh6fuHXO4fRSJoJhUVCOX3G1qjld2D0mtLlX_o9oMnq2WNXEPLdyb9Wf4k-1Z_fVhxsUvV5Y79LZ2fpOD4445V40lUdKfbkYM0cGjIdl96-SzrU2yTfvVTQ.png ? 

I think a step by step installation guide would be helpful here, with the screenshots as supplement resources.

File: pages/inspector/connect-inspector-to-gtm.mdx [hunk 6]
```diff
@@ -1,14 +1,74 @@
...
-You can use a server side GTM to analyze your tracking with Avo Inspector.
+_This document describes how to send data from Google Tag Manager to Avo Inspector._
 
-We provide a tag template to import, which you can use to create a tag and connect it to your GTM clients.
+<Callout type="info">
+The Inspector Google Tag Manager integrations are currently in beta so please reach out to us at support@avo.app or in the chat bubble in the bottom right corner if you have any feedback.
+</Callout>
 
-> Avo Inspector operates on the environment level, please keep it in mind and create separate tags for development, staging and production data.
+## Destination Info
```
Comment (rhys-growth-avo) [inline pages/inspector/connect-inspector-to-gtm.mdx:11]: @thoragudf think this should be Source Info, no?

File: pages/inspector/connect-inspector-to-gtm.mdx [hunk 7]
```diff
@@ -1,14 +1,74 @@
...
 
-## Guide to connect GTM and Avo Inspector
+* Supports both client side and server side Google Tag Manager  
+* Inspector events can be verified using the browser developer console
```
Comment (rhys-growth-avo) [inline pages/inspector/connect-inspector-to-gtm.mdx:14]: @alex-tpom6oh how exactly can we do this? Maybe we want to specify the key needed to find them?

File: pages/inspector/connect-inspector-to-gtm.mdx [hunk 8]
```diff
@@ -1,14 +1,74 @@
...
+* Observability can be added to all data layer events using client side GTM in under 90 seconds
+* Observability can be added to all Google Analytics events using our server side GTM installation option
+* Events detected by Inspector can be added to the Avo tracking plan within the Inspector UI, allowing you to fill a documentation gap on legacy events in seconds
+
+
+Google Tag Manager (GTM) is a tag management system that allows you to quickly update tags and code snippets on your website, web apps, and apps. Once you deploy the Tag Manager snippet, you can configure (trigger) tags using a web-based user interface without having to alter and deploy additional code. It is common for early stage companies to use Google Tag Manager to take care of dispatching their core analytics events to reduce developer dependencies whenever new events are needed, and as the analytics maturity grow and more accuracy is needed, developers will deploy in-code data layer events, and then GTM tag owners will trigger the analytics service events off the data layer events accordingly. [Learn more about that here](https://www.loom.com/share/22c9e0f1175141efa7a0bdbf51201096).
```
Comment (rhys-growth-avo) [inline pages/inspector/connect-inspector-to-gtm.mdx:20]: @thoragudf please replace with 

Google Tag Manager (GTM) is a tag management system that allows you to quickly update tags and code snippets on your website, web apps, and apps. Once you deploy the Tag Manager snippet, you can configure (trigger) tags using a web-based user interface without having to alter and deploy additional code. It is common for early stage companies to use Google Tag Manager to take care of dispatching their core analytics events to reduce developer dependencies whenever new events are needed. Typically, as the analytics maturity grow and more accuracy is needed, developers will deploy data layer events in their codebase, and then GTM tag owners will trigger the analytics service events from these data layer events. [Learn more about that here](https://www.loom.com/share/22c9e0f1175141efa7a0bdbf51201096).

File: pages/inspector/connect-inspector-to-gtm.mdx [hunk 9]
```diff
@@ -1,14 +1,74 @@
...
+* Observability can be added to all data layer events using client side GTM in under 90 seconds
+* Observability can be added to all Google Analytics events using our server side GTM installation option
+* Events detected by Inspector can be added to the Avo tracking plan within the Inspector UI, allowing you to fill a documentation gap on legacy events in seconds
+
+
+Google Tag Manager (GTM) is a tag management system that allows you to quickly update tags and code snippets on your website, web apps, and apps. Once you deploy the Tag Manager snippet, you can configure (trigger) tags using a web-based user interface without having to alter and deploy additional code. It is common for early stage companies to use Google Tag Manager to take care of dispatching their core analytics events to reduce developer dependencies whenever new events are needed, and as the analytics maturity grow and more accuracy is needed, developers will deploy in-code data layer events, and then GTM tag owners will trigger the analytics service events off the data layer events accordingly. [Learn more about that here](https://www.loom.com/share/22c9e0f1175141efa7a0bdbf51201096). 
+
+**_Reminders:_**
```
Comment (rhys-growth-avo) [inline pages/inspector/connect-inspector-to-gtm.mdx:22]: @thoragudf can we make in one of those colour highlighted boxes like we have done elsewhere in this doc, or you think that's a worst experience?

File: pages/inspector/connect-inspector-to-gtm.mdx [hunk 10]
```diff
@@ -1,14 +1,74 @@
...
+
+**_Reminders:_** 
+- The Inspector SDKs are designed with flexibility in mind, so you should always be able to add observability to the source of your pipelines.
+- Avo Inspector operates on the environment level, please keep it in mind and create separate tags for development, staging and production data.
+
+## Getting Started
+Avo Inspector requires two core steps to add observability:
+1. Initializing the Inspector
+2. Sending events to Inspector
```
Comment (rhys-growth-avo) [inline pages/inspector/connect-inspector-to-gtm.mdx:29]: @thoragudf Replace with

Sending your events to Inspector

File: pages/inspector/connect-inspector-to-gtm.mdx [hunk 11]
```diff
@@ -1,14 +1,74 @@
...
+There are two options to install Inspector via Google Tag Manager. You can either import our GTM recipe to add observability to all data layer events, or use our Inspector function to add observability to individual events.
+
+<Callout type="info" emoji="💡">
+We recommend starting by installing the Inspector on your development or testing GTM container and connect it to the Inspector dev environment. That is both so your Inspector production environment is not contaminated by testing data and because the Inspector development environment is designed to handle low data volumes real time while the production environment is optimized for large amount of production data and has 1 - 2 hour delay.
+</Callout>
+
+### Importable GTM recipe
+
+To add observability to all data layer events with our importable GTM recipe [in under 90 seconds](https://www.loom.com/share/c659b857d4d947729683bf11691f441e), follow the steps below:
+1. [Download our GTM recipe here]()
```
Comment (rhys-growth-avo) [inline pages/inspector/connect-inspector-to-gtm.mdx:44]: @thoragudf if Alex signs off the recipe after my last comment in Asana, we need to host the code somewhere they can download it when clicking this link.

File: pages/inspector/connect-inspector-to-gtm.mdx (hunk 1 above)
Comment (rhys-growth-avo) [reply pages/inspector/connect-inspector-to-gtm.mdx:45]: https://drive.google.com/file/d/1cjfox9gi3lTYC5uFJemWlwdpInL2s8I3/view?usp=drive_link. 

+ reword to 

2. Import the recipe into your GTM container. The recipe contains the following tags and variables.
Tags
- Avo Inspector Init
- Avo Inspector Send

Variables
- Avo Inspector API Key 

// @alex-tpom6oh have we removed the gtm.uniqueEventId in the new snippit?)

File: pages/inspector/connect-inspector-to-gtm.mdx (hunk 2 above)
Comment (rhys-growth-avo) [reply pages/inspector/connect-inspector-to-gtm.mdx:46]: https://drive.google.com/file/d/1ILy8gVRe2jVffn4ZN4p9zA1uHIDurP2R/view?usp=drive_link

File: pages/inspector/connect-inspector-to-gtm.mdx (hunk 3 above)
Comment (rhys-growth-avo) [reply pages/inspector/connect-inspector-to-gtm.mdx:47]: https://drive.google.com/file/d/1qeHpVlznE_XwL5rNakKK-B0wnMLlSpGu/view?usp=drive_link

CC @alex-tpom6oh should we instead make the init tag reference a dynamic variable... and then just show a screen shot of setting the variable. Would this still work?

File: pages/inspector/connect-inspector-to-gtm.mdx (hunk 4 above)
Comment (rhys-growth-avo) [reply pages/inspector/connect-inspector-to-gtm.mdx:49]: Agreed.

File: pages/inspector/connect-inspector-to-gtm.mdx [hunk 12]
```diff
@@ -1,14 +1,74 @@
...
+2. Import the recipe into your GTM container. 
+3. Add the source specific Inspector API key. 
+4. When testing, adjust the the Avo Inspector Init tag by changing the 'prod' value to 'dev' on the inspector.__ENV__ key.
+5. Publish the container.
+6. Visit the Avo Inspector dashboard in one hour to see
+7. Consider creating a filtered issues view for the data bugs you care about. [learn more here](/inspector/inspector-issues-view).
+
+<Callout type="info" emoji="💡">
```
Comment (rhys-growth-avo) [inline pages/inspector/connect-inspector-to-gtm.mdx:52]: @thoragudf think we should move this to the top of this section. inItialisation is like a step one. Ideally we have a whole page section on Initialisation, but easy to get into a rabbit whole there, so kept this GTM focused. But bumping it to the top helps skimmers know that this is baked into the recipe.

Comment (rhys-growth-avo) [reply pages/inspector/connect-inspector-to-gtm.mdx:52]: I mean sat under this subheader https://docs-git-inspector-web-gtm-beta.avo-dev.app/docs/inspector/connect-inspector-to-gtm#importable-gtm-recipe

File: pages/inspector/connect-inspector-to-gtm.mdx (hunk 5 above)
Comment (rhys-growth-avo) [reply pages/inspector/connect-inspector-to-gtm.mdx:60]: Thats the correct image. That image should be enough for now. I think we leave a more fleshed out guide for a Timo video. That image would have been enough for me, even when I was just starting out with GTM

File: pages/inspector/connect-inspector-to-gtm.mdx [hunk 13]
```diff
@@ -1,14 +1,74 @@
...
+The Initialization code is already included in a tag, so you just need to provide the Inspector API GTM variable the source specific API key, which you can find in your Avo source Inspector setup. 
+</Callout>
+
+### Inspector Get Schema function
+
+Another option is to initialize Inspector via creating your own HTML tag and using the HTML script you can find in the source Inspector setup.
```
Comment (rhys-growth-avo) [inline pages/inspector/connect-inspector-to-gtm.mdx:58]: @thoragudf hmmm. This feels out if place. I wonder if we need an inItialisation sub header in getting started section. And explain that if you are using our recipe, it's baked in. If you are not, you can grab the code from Avo source setup and create a HTML tag or add it to the header of your website. WDYT?

File: pages/inspector/connect-inspector-to-gtm.mdx [hunk 14]
```diff
@@ -1,14 +1,74 @@
...
+
+To add observability to single events with our Inspector Get Schema function, create a HTML script tag for each event you want to add observability to. The steps are shown in the screen shot below and make sure to use the same trigger and variables as you are using for the event you're dispatching to your analytics vendor. 
+
+<Callout type="info" emoji="💡">
+If you have occasions where data layer events could fire before GTM loads and you wish you add observability, we'd recommend initializing inspector outside of Google Tag Manager, and perhaps adding our initialization code to the header of the page or using one of our other Inspector SDKs and adding it more native to your codebase. The flexibility here is a feature.
```
Comment (rhys-growth-avo) [inline pages/inspector/connect-inspector-to-gtm.mdx:63]: @thoragudf typo. 

Change "fire before GTM loads and you wish you add observability,"

to 

fire before GTM loads and you wish to add observability,

File: pages/inspector/connect-inspector-to-gtm.mdx [hunk 15]
```diff
@@ -1,14 +1,74 @@
...
+</Callout>
+
+[insert image from our latest blog on GTM]
+
+> **_Notes:_** 
+While our importable recipe allows you to add observability to all data layer events in production in under 90 seconds, we are aware that many GTM containers are a hybrid of data layer based triggered tags for analytics events, and GTM native ones (think source selectors). If your GTM analytics implementation uses a hybrid trigger setup, to increase Inspector coverage, you could use our recipe to catch all data layer events with a single tag, and then single event Inspector Tags for native GTM trigger based events. If you wanted a single tag that catches both data layer and native GTM triggered events sent Google Analytics 4, for example, ask your developer to code a tag that detects all analytics track calls via the GA4 end point calls, parses the payload, and loops through the properties to then forward the track on to Inspector. We're currently investigating this for our new recipe. 
+
```
Comment (rhys-growth-avo) [inline pages/inspector/connect-inspector-to-gtm.mdx:70]: @thoragudf maybe we add a feature request link here, and mention if you want to get notified, upvote this feature request. WDYT?

**Comment by thoragudf** [comment]
Updated PR according to the environment setup changes

File: pages/inspector/connect-inspector-to-gtm.mdx [hunk 16]
```diff
@@ -1,14 +1,106 @@
...
+Go to the Google Tag Manager container you want to install the Inspector on and import the recipe.
+
+![Import screen in Google Tag Manager](/images/inspector/import-gtm-recipe-to-container.png)
+
+The recipe contains the following tags and variables:
+- Tags
+    - Avo Inspector Init
+    - Avo Inspector Send
+- Variables
+    - Avo Inspector API Key
```
[the remark this answers is not in the document]
Comment (thoragudf) [reply pages/inspector/connect-inspector-to-gtm.mdx:54]: Is it correct that the only thing that needs to be touched is the Avo Inspector API key @alex-tpom6oh ?

File: pages/inspector/connect-inspector-to-gtm.mdx [hunk 17]
```diff
@@ -1,14 +1,99 @@
...
+
+
+Google Tag Manager (GTM) is a tag management system that allows you to quickly update tags and code snippets on your website, web apps, and apps. Once you deploy the Tag Manager snippet, you can configure (trigger) tags using a web-based user interface without having to alter and deploy additional code. It is common for early stage companies to use Google Tag Manager to take care of dispatching their core analytics events to reduce developer dependencies whenever new events are needed. Typically, as the analytics maturity grow and more accuracy is needed, developers will deploy data layer events in their codebase, and then GTM tag owners will trigger the analytics service events from these data layer events. [Learn more about that in this short video](https://www.loom.com/share/22c9e0f1175141efa7a0bdbf51201096).
+
+<Callout type="info">
+**_Reminders:_** 
+- The Inspector SDKs are designed with flexibility in mind, so you should always be able to add observability to the source of your pipelines.
+- Avo Inspector operates on the environment level, please keep it in mind and create separate tags for development, staging and production data.
```
[the remark this answers is not in the document]
Comment (thoragudf) [reply pages/inspector/connect-inspector-to-gtm.mdx:25]: yes!

**Pull request merged**